mcs/class/Mono.Btls.Interface/Mono.Btls.Interface/BtlsX509.cs

```diff
@@ -87,7 +87,10 @@ public byte[] GetPublicKeyData ()
 
 		public byte[] GetSerialNumber (bool mono_style)
 		{
-			return Instance.GetSerialNumber (mono_style);
+			var serial = Instance.GetSerialNumber (mono_style);
+			if (mono_style)
+				Array.Reverse (serial);
+			return serial;
```

Comment:
What does `mono_style` mean? I was under the impression that it means "little endian", but then I don't understand why it would be both passed to the underlying implementation and then `Array.Reverse`d.

Reply:
`MonoBtlsX509.GetSerialNumber(false)` returns the raw data as they're stored in the certificate.

`MonoBtlsX509.GetSerialNumber(true)` returns it in the format expected by CoreFX's PAL - which is big-endian plus padding.  This is not a direct reverse of the first and we have a test somewhere which exposes the difference.

The `Array.Reverse()` here in `Mono.Btls.Interface` (which is used by the web-tests) was added because the internal implementation (in `MonoBtlsX509`) switched from returning little-ending to big-ending (to match CoreFX PAL) - to avoid regressing the web-tests with a behavior change in the "semi-public" `Mono.Btls.Interface` code.

Reply:
We could also rename `mono_style` into `dotnet_style` as it is now the same.